pkgs/unified_analytics/lib/src/enums.dart

```diff
@@ -141,6 +144,18 @@ enum DashTool {
     required this.label,
     required this.description,
   });
+
+  /// This takes in the string label for a given [DashTool] and returns the
+  /// enum for that string label.
+  static DashTool getDashToolByLabel(String label) {
+    for (final tool in DashTool.values) {
+      if (tool.label == label) return tool;
+    }
+
+    throw Exception('The tool $label from the survey metadata file is not '
+        'a valid DashTool enum value\n'
+        'Valid labels for dash tools: ${validDashTools.join(', ')}');
```

Comment:
It looks to me like this exception will be thrown when we're attempting to parse the survey file from the server. I don't think we want to throw an exception at that point. I suspect that we want to either ignore the invalid tool label, ignore the survey that contains it, or ignore all of the surveys. The caller (whether a CLI tool or the analysis server) shouldn't need to handle this exception, nor should they crash if they forget to handle it.

What we probably do need is a validation test that we can run over the survey file before we upload it to the server to ensure that it doesn't contain invalid data in the first place.

Reply:
Yeah if there is an error when parsing the json for a given survey, the plan is to not surface this to the user at all, and instead just fail silently. This snippet below shows that we are catching and returning `null` for any errors we may anticipate

https://github.com/eliasyishak/tools/blob/85558638ad791aee6f99c448feabfd5ec0cbae0a/pkgs/unified_analytics/lib/src/survey_handler.dart#L313-L336

So if we have trouble getting a valid `DashTool`, the entire survey will be ignored.

Additionally, we have [ci workflows](https://github.com/flutter/uxr/blob/master/surveys/survey-validator/lib/survey_validator.dart) in the `flutter/uxr` to prevent this kind of error before we even publish the survey

Reply:
Sounds like my concerns here were already covered. Thanks.